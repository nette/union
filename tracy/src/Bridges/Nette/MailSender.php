<?php

/**
 * This file is part of the Tracy (https://tracy.nette.org)
 * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
 */

namespace Tracy\Bridges\Nette;

use Nette;
use Tracy\Dumper;
use Tracy\Helpers;


/**
 * Tracy logger bridge for Nette Mail.
 */
class MailSender
{
	use Nette\SmartObject;

	/** @var Nette\Mail\IMailer */
	private $mailer;

	/** @var string|null sender of email notifications */
	private $fromEmail;


	public function __construct(Nette\Mail\IMailer $mailer, $fromEmail = null)
	{
		$this->mailer = $mailer;
		$this->fromEmail = $fromEmail;
	}


	/**
	 * @param  string|\Exception|\Throwable  $message
	 * @param  string  $email
	 * @return void
	 */
	public function send($message, $email)
	{
		$host = preg_replace('#[^\w.-]+#', '', isset($_SERVER['HTTP_HOST']) ? $_SERVER['HTTP_HOST'] : php_uname('n'));

		$mail = new Nette\Mail\Message;
		$mail->setHeader('X-Mailer', 'Tracy');
		$mail->setFrom($this->fromEmail ?: "noreply@$host");
		$mail->addTo($email);
		$mail->setSubject('PHP: An error occurred on the server ' . $host);
		$mail->setBody(static::formatMessage($message) . "\n\nsource: " . Helpers::getSource());

		$this->mailer->send($mail);
	}


	/**
	 * @param  mixed  $message
	 * @return string
	 */
	private static function formatMessage($message)
	{
		if ($message instanceof \Exception || $message instanceof \Throwable) {
			while ($message) {
				$tmp[] = ($message instanceof \ErrorException
					? Helpers::errorTypeToString($message->getSeverity()) . ': ' . $message->getMessage()
					: Helpers::getClass($message) . ': ' . $message->getMessage() . ($message->getCode() ? ' #' . $message->getCode() : '')
				) . ' in ' . $message->getFile() . ':' . $message->getLine();
				$message = $message->getPrevious();
			}
			$message = implode("\ncaused by ", $tmp);

		} elseif (!is_string($message)) {
			$message = Dumper::toText($message);
		}

		return trim($message);
	}
}
