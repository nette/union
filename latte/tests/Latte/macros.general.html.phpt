<?php

/**
 * Test: Nette\Latte\Engine: general HTML test.
 *
 * @author     David Grudl
 */

use Nette\Latte,
	Nette\Utils\Html,
	Tester\Assert;


require __DIR__ . '/../bootstrap.php';


$latte = new Latte\Engine;
$latte->setTempDirectory(TEMP_DIR);
$latte->addFilter('translate', 'strrev');
$latte->addFilter('join', 'implode');

$params['hello'] = '<i>Hello</i>';
$params['xss'] = 'some&<>"\'/chars';
$params['people'] = array('John', 'Mary', 'Paul', ']]> <!--');
$params['menu'] = array('about', array('product1', 'product2'), 'contact');
$params['el'] = Html::el('div')->title('1/2"');

$path = __DIR__ . '/expected/' . basename(__FILE__, '.phpt');
Assert::matchFile(
	"$path.phtml",
	$latte->compile(__DIR__ . '/templates/general.latte')
);
Assert::matchFile(
	"$path.html",
	$latte->renderToString(
		__DIR__ . '/templates/general.latte',
		$params
	)
);
