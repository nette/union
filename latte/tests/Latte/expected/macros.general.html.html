<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<ul>
	<li>Escaped: &lt;i&gt;Hello&lt;/i&gt;</li>
	<li>Non-escaped: <i>Hello</i></li>
	<li>Escaped expression: &lt;b&gt;hello&lt;/b&gt;</li>
	<li>Non-escaped expression: <b>hello</b></li>
	<li>Array access: Mary</li>
	<li>Html: <div title='1/2"'></div></li>
	<li>Translated: yraM</li>
	<li>Translated string: joha</li>
	<li>Non-escaped and translated: yraM</li>
	<li>Condition: yes yes</li>
	<li>Array: abc abc</li>
</ul>

<input> <input> <input>

<input checked> <input checked="checked">

{ test} {"test} {'test}

<p val = "a"> </p>

<p val0 val1 = tmp val2=tmp val3="x"></p> </p>

<p val = "some&amp;&lt;&gt;&quot;&#039;/chars" val2="`mxss "> </p>

<p onclick = "some&amp;&lt;&gt;&quot;&#039;/chars"> </p>

<p ONCLICK ="&quot;some&amp;&lt;&gt;\&quot;&#039;\/chars&quot;" "some&amp;&lt;&gt;&quot;&#039;/chars"> </p>

<p val = />some&amp;&lt;&gt;"'/chars</p>

1

	&lt;i&gt;Hello&lt;/i&gt;


	<ul>
	<li id="item-1" class="odd">John</li>


	<li id="item-2" class="even">Mary</li>


	<li id="item-3" class="odd">Paul</li>


	<li id="item-4" class="even">]]&gt; &lt;!--</li>
	</ul>



<STYLE type="text/css">
<!--
#some\&\<\>\"\'\/chars {
	background: blue;
}
-->
</style>



<script>
<!--
alert('</div>');

var prop = ["John","Mary","Paul","]]\x3E \x3C!--"];

document.getElementById("some&<>\"'\/chars").style.backgroundColor = 'red';

var html = "<div title='1\/2\"'><\/div>";
-->
</script>


<SCRIPT>
/* <![CDATA[ */

var prop2 = ["John","Mary","Paul","]]\x3E \x3C!--"];

/* ]]> */
</script>


<p onclick =
'alert(&quot;some&amp;&lt;&gt;\&quot;&#039;\/chars&quot;);alert("hello");'
 title='some&amp;&lt;&gt;&quot;&#039;/chars'
 STYLE =
 "color:some\&amp;\&lt;\&gt;\&quot;\&#039;\/chars;"
 rel="some&amp;&lt;&gt;&quot;&#039;/chars"
 onblur="alert(&quot;some&amp;&lt;&gt;\&quot;&#039;\/chars&quot;)"
 alt='&lt;div title=&#039;1/2&quot;&#039;&gt;&lt;/div&gt;'
 onfocus="alert(&quot;&lt;div title=&#039;1\/2\&quot;&#039;&gt;&lt;\/div&gt;&quot;)"
>click on me some&amp;&lt;&gt;"'/chars</p>



<ul>
	<li>0 about</li>
	<li>1  <ul>
	<li>2 product1</li>
	<li>3 product2</li>
</ul>
 </li>
	<li>2 contact</li>
</ul>


2
2

8
9




</ul>


test

{ }

true

true


y

<br>



<button></button>

<button></button> xx
